authorization/src/main/java/io/deephaven/auth/ServiceAuthWiring.java

```diff
@@ -0,0 +1,194 @@
+                FunctionalInterfaces.ThrowingRunnable<ReflectiveOperationException> authRunner) {
+            try {
+                authRunner.run();
+            } catch (ReflectiveOperationException | RuntimeException originalErr) {
+                Throwable err = originalErr;
+                if (!(err instanceof StatusRuntimeException) && err.getCause() != null) {
+                    err = err.getCause();
+                }
+                if (err instanceof StatusRuntimeException) {
+                    final StatusRuntimeException sre = (StatusRuntimeException) err;
+                    Status.Code status = sre.getStatus().getCode();
+                    switch (status) {
+                        case UNAUTHENTICATED:
+                            log.info().append(serviceName).append(".").append(methodName)
+                                    .append(": request unauthenticated").endl();
+                            break;
+                        case PERMISSION_DENIED:
+                            log.info().append(serviceName).append(".").append(methodName)
+                                    .append(": request unauthorized").endl();
+                            break;
+                        case RESOURCE_EXHAUSTED:
+                            log.info().append(serviceName).append(".").append(methodName)
+                                    .append(": request throttled").endl();
+                            break;
+                        default:
+                            log.error().append(serviceName).append(".").append(methodName)
+                                    .append(": authorization failed: ").append(err).endl();
+                    }
+
+                    quietlyCloseCall(call, sre.getStatus(), sre.getTrailers());
+                    return false;
+                }
+
+                // this is an unexpected error from the auth wiring invocation, be loud about it
+                log.error().append("Failed to invoke auth method: ").append(originalErr).endl();
+                quietlyCloseCall(call);
+                return false;
+            }
+
+            return true;
+        }
+    }
+
+    /**
+     * Close the call, but don't throw any exceptions. We were not given details about the error.
+     */
+    private static <ReqT, RespT> void quietlyCloseCall(final ServerCall<ReqT, RespT> call) {
+        quietlyCloseCall(call, Status.UNAUTHENTICATED, new Metadata());
+    }
+
+    /**
+     * Close the call, but don't throw any exceptions.
+     */
+    private static <ReqT, RespT> void quietlyCloseCall(
+            final ServerCall<ReqT, RespT> call, final Status status, final Metadata trailers) {
+        try {
+            call.close(status, trailers);
+        } catch (IllegalStateException ignored) {
+            // could be thrown if the call was already closed. As an interceptor, we can't throw,
+            // so ignoring this and just returning the no-op listener.
```

Comment:
comment was copied, but not quite accurate, there is no no-op listener here (though both cases that call this return false, which right away will return a no-op listener)